### ISSUE TITLE
languageserver features doesn't work on saved files

### ISSUE THREAD

**Issue by zamirD123** [REditorSupport/languageserver#595]
Hi all,
I've noticed that lsp featruers as autocompliation, on hover popup etc aren't working when I'm working on a saved R file, but if I start a new file and set it as an R file (without seving) everything works fine. 

**Describe the bug**
autocompliation, and on hover popups aren't working when editing saved files, but do work on an unsaved R file.

**To Reproduce**
1. Open a saved .R file
2. Type a function name or hover over a function ,and expact popup to apear
3. nothing happens
4. Open a new file (`ctrl`+`alt`+ `win` +`n`),  choose  `R document` from the dropdown
5. Start typing a function name or hover over a fucntion: a pop up appears!

User `settings.JSON` :
![image](https://user-images.githubusercontent.com/69770500/214544302-9030a93f-61c0-4a55-adf5-65d9b146f2de.png)

Workspace `settings.JSON`:
![image](https://user-images.githubusercontent.com/69770500/214544418-1c0c5159-1800-4f60-b69f-12123daf5c15.png)

I'm using Win 11, and a fresh install of VScode, R extention and `languageserver`

**Comment by ratnanil** [comment]
OMG I'm so glad there is already an open issue on this. I am experiencing exactly the same issue. 

### Autocomplete in an *un*saved file:

Note how I can autocomplete function arguments using tab.

![unsaved-file](https://user-images.githubusercontent.com/12532091/215094641-6ddf25fa-dac3-46e8-9259-5d906cbc423f.gif)


### Autocomplete in an saved file:

The range of functions appearing when I type `rno` is very much reduced, `rnorm` is not available. Also autocompleting function arguments is now not possible.


![saved-file](https://user-images.githubusercontent.com/12532091/215094871-1c59cae3-c955-4b6c-a928-a32f7c275349.gif)

**Comment by zamirD123** [comment]
@ratnanil thank you! This is exactly what I am experiencing

**Comment by zamirD123** [comment]
Anyone?

**Comment by renkun-ken** [comment]
Would you like to turn on `r.lsp.debug` and go to Output -> R Language Server and see if there is something wrong in the log?

**Comment by zamirD123** [comment]
Hi @renkun-ken, This is with debug on (nothing is happaning when i'm hovering over code, etc.)
![image](https://user-images.githubusercontent.com/69770500/218015410-fea94a4c-9013-43ac-8ae6-9f1f7818b967.png)

**Comment by renkun-ken** [comment]
Do you mean that when you edit the code or hover in this saved file, no log is produced by the language server?

Looks like the language client ignores this file for some reason. What's the full path of this file?

**Comment by zamirD123** [comment]
That's true. no log is pruduced and it seems like the client ignores any saved file...
That's the file path: `C:\Alternativ\scripts\analysis\public_transit_changes\tactic_nataz_bus_speeds_analysis.R`

**Comment by renkun-ken** [comment]
What's the path of your workspace folder?

**Comment by zamirD123** [comment]
I'm not sure how to find it.. I think It's C:/
[image: image.png]
